koku/masu/processor/report_parquet_processor_base.py

```diff
@@ -124,18 +124,29 @@ def _generate_create_table_sql(self):
             sql += f"{norm_col} {col_type}"
             if idx < (len(parquet_columns) - 1):
                 sql += ","
-        sql += ",source varchar, year varchar, month varchar"
+        if partition_map:
+            # Add the specified partition columns
+            sql += ", "
```

Comment:
I know this is continuing an existing pattern so I wouldn't change it, but string concatenation using `+` and `+=` can be quite expensive since each intermediate string needs to be built before assignment.

Collecting these values in a mapping or iterable and doing a single `f` string or `.join()` at the end is more efficient and doesn't sacrifice legibility.

Reply:
This is a good note for a potential refactor of some of this, I think for now I will leave it as is per the comment.